variables.tf

```diff
@@ -71,20 +88,64 @@ variable "pubsub_maximum_backoff" {
   default     = "600s"
 }
 
+variable "function_roles" {
+  description = <<-EOF
+    A list of IAM roles to give the Cloud Function.
+  EOF
+  type        = list(string)
+
+  default = [
+    "roles/compute.viewer",
+    "roles/iam.serviceAccountViewer",
+    "roles/cloudscheduler.viewer",
+    "roles/cloudasset.viewer",
+    "roles/browser", # for viewing projects
+  ]
+}
+
+variable "enable_function" {
+  description = "Whether to enable the Cloud function"
+  type        = bool
+  default     = true
+}
+
+variable "function_bucket" {
+  description = "GCS bucket containing the Cloud Function source code"
+  type        = string
+  default     = "observeinc"
+}
+
+variable "function_object" {
+  description = "GCS object key of the Cloud Function source code zip file"
+  type        = string
+  default     = "google-cloud-functions.zip"
+}
+
+variable "function_schedule" {
+  description = <<-EOF
+    How often to trigger the cloud function. This is a Cloud Scheduler Job schedule:
+    https://cloud.google.com/scheduler/docs/reference/rest/v1/projects.locations.jobs#Job
+  EOF
+  type        = string
+  default     = "*/5  * * * *"
+}
+
+# tflint-ignore: terraform_unused_declarations
 variable "project_id" {
```

Comment:
I would think this shouldn't be left in, given that it's not really deprecated, it's not referenced at all, meaning that it doesn't work. For it to be deprecated, it would need to still work. I think it should be possible to pass this to `data.google_project.this`, in combination with my `null` suggestion below, and maintain compatibility. 

FWIW, I agree with expecting the provider instance to set the project/region rather than accepting that and passing it to every resource. It would be reasonable to cut a breaking release for this.

Reply:
Yeah, I guess it's better to delete it. it will mean the user needs to change some things on there end, but shouldn't be too bad

Reply:
this reminds me we should have an example of how to set up multiple projects (using multiple providers)